File: PorkChop.py
# MDI: Version 0.2

import RPi.GPIO as GPIO
import requests
import time

# Set the GPIO pin for the door status input
GPIO_PIN = 17  # Replace with your actual GPIO pin number

# Google Form URL
FORM_URL = "https://docs.google.com/forms/d/e/1FAIpQLSft-dOSr_axCy-EMB6yN1gXn1ELDZqXTfrGCmcUCmksbaaAlg/viewform"  # Replace with your Google Form URL

def update_door_status(status):
    data = {"entry.1565515877": status}  # Replace INSERT_QUESTION_ID_HERE with the actual question ID
    response = requests.post(FORM_URL, data=data)
    if response.status_code == 200:
        print("Form updated successfully")
    else:
        print("Failed to update form")
	print("Response Status Code:", response.status_code)
	print("Post data: ", data)

try:
    # Set up GPIO
    GPIO.setmode(GPIO.BCM)
    GPIO.setup(GPIO_PIN, GPIO.IN)

    while True:
        if GPIO.input(GPIO_PIN) == GPIO.LOW:
            # GPIO contact is closed, update the form with "DoorClosed"
            update_door_status("DoorClosed")
        else:
            # GPIO contact is open, update the form with "DoorOpen"
            update_door_status("DoorOpen")

        time.sleep(60)  # Update status every minute (adjust as needed)
except KeyboardInterrupt:
    GPIO.cleanup()

File: PorkChop.js
// Replace this with your Google Form URL
const FORM_URL = 'https://docs.google.com/forms/d/e/1FAIpQLSft-dOSr_axCy-EMB6yN1gXn1ELDZqXTfrGCmcUCmksbaaAlg/viewform';
// Need to replace these with actual image file paths
const RED_IMAGE_PATH = 'red.png';
const GREEN_IMAGE_PATH = 'green.png';

function updateStatusIndicator() {
    // Fetch the Google Form responses
    fetch(FORM_URL)
        .then((response) => response.text())
        .then((data) => {
            // Parse the HTML content of the Google Form
            const parser = new DOMParser();
            const doc = parser.parseFromString(data, 'text/html');

            // Find the question's response in the HTML (replace 'question-id' with the actual question's ID ****Where at?****  But ID is entry.1565515877)
            const questionResponse = doc.querySelector('.freebirdFormviewerViewItemsRadioChoice[aria-checked="true"]');

            // Find the indicator image element
            const indicatorImg = document.getElementById('indicator-img');

            if (questionResponse) {
                // Set the image based on the response
                if (questionResponse.innerText === 'Closed') {
                    indicatorImg.src = RED_IMAGE_PATH;
                } else if (questionResponse.innerText === 'Open') {
                    indicatorImg.src = GREEN_IMAGE_PATH;
                }
            }
        })
        .catch((error) => {
            console.error('Failed to fetch form responses:', error);
        });
}

// Call the updateStatusIndicator function to update the indicator on page load
updateStatusIndicator();
// Poll for updates every 10 seconds (adjust as needed)
setInterval(updateStatusIndicator, 10000);
